src/test/java/uk/gov/justice/probation/courtcasematcher/service/TelemetryServiceTest.java

```diff
@@ -0,0 +1,248 @@
+import uk.gov.justice.probation.courtcasematcher.model.offendersearch.SearchResponse;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.data.MapEntry.entry;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static uk.gov.justice.probation.courtcasematcher.service.TelemetryService.CASE_NO_KEY;
+import static uk.gov.justice.probation.courtcasematcher.service.TelemetryService.COURT_CODE_KEY;
+import static uk.gov.justice.probation.courtcasematcher.service.TelemetryService.COURT_ROOM_KEY;
+import static uk.gov.justice.probation.courtcasematcher.service.TelemetryService.CRNS_KEY;
+import static uk.gov.justice.probation.courtcasematcher.service.TelemetryService.DATE_OF_HEARING_KEY;
+import static uk.gov.justice.probation.courtcasematcher.service.TelemetryService.MATCHED_BY_KEY;
+import static uk.gov.justice.probation.courtcasematcher.service.TelemetryService.MATCHES_KEY;
+import static uk.gov.justice.probation.courtcasematcher.service.TelemetryService.PNC_KEY;
+
+@DisplayName("Exercise TelemetryService")
+@ExtendWith(MockitoExtension.class)
+class TelemetryServiceTest {
+
+    private static final String COURT_CODE = "B10JQ00";
+    private static final String CASE_NO = "1234567890";
+    private static final String CRN = "D12345";
+    private static final String COURT_ROOM = "01";
+    private static final LocalDate DATE_OF_HEARING = LocalDate.of(2020, Month.NOVEMBER, 5);
+    private static Info info;
+    private static Case aCase;
+
+    @Captor
+    private ArgumentCaptor<Map<String, String>> propertiesCaptor;
+
+    @Mock
+    private TelemetryClient telemetryClient;
+
+    @InjectMocks
+    private TelemetryService telemetryService;
+
+
+    @BeforeAll
+    static void beforeEach() {
+
+        info = Info.builder()
+            .infoSourceDetail(InfoSourceDetail.builder().ouCode(COURT_CODE).build())
+            .dateOfHearing(DATE_OF_HEARING)
+            .build();
+        Document document = Document.builder().info(info).build();
+        DataJob dataJob = DataJob.builder().document(document).build();
+        Job job = Job.builder().dataJob(dataJob).build();
+
+        Session session = Session.builder()
+            .courtName("COURT_NAME")
+            .courtRoom(COURT_ROOM)
+            .dateOfHearing(DATE_OF_HEARING)
+            .job(job)
+            .build();
+
+        aCase = Case.builder()
+            .block(Block.builder()
+                .session(session)
+                .build())
+            .caseNo(CASE_NO)
+            .build();
+    }
+
+    @DisplayName("Simple record of event with no properties")
+    @Test
+    void whenMessageReceived_thenRecord() {
+        telemetryService.trackEvent(TelemetryEventType.COURT_LIST_RECEIVED);
+
+        verify(telemetryClient).trackEvent("PiCCourtListReceived");
+    }
+
+    @DisplayName("Record the event when an exact match happens")
```

Comment:
This is more an observation than anything but the approach I took with the court-case-service TelemetryService was to test it in a way that's agnostic to the actual event being tracked. So as a service it's expected to be able to take an event type and an object and produce telemetry based on the contents of that object. The advantage being that you only really need to test two cases per method - the one where everything is present, and where the minimal subset is present. It's then up to the calling service to ensure that the right data is being sent, giving a clean separation of responsibilities.

Reply:
I see what you mean. I think that you have taken the approach of passing the type of event to the service, presumably making the decision about which type is required  outside the method.Presumably that has meant more complexity in the calling services ?

Reply:
Yes marginally, though it's just one additional argument. 

What I hadn't clocked about this use case is that the objects being acted upon are different in each case - whereas in the court-case-service we have multiple events with the same object being logged so you gain a benefit from being able to reuse the methods. Given none of the methods here are reusable anyway it doesn't make much sense to do it the way I did it in court-case-service. 

I commented more out of curiosity than anything as I suspected there would be a good reason why we did it differently!